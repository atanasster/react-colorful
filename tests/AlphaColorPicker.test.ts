import * as path from "path";
import { loadConfigurations } from "@component-controls/config";
import { renderExample } from "@component-controls/test-renderers";
import { render, act } from "@testing-library/react";
import { renderErr } from "@component-controls/test-renderers";

import doc, { Hex, Rgba } from "./../docs/src/common/AlphaColorPicker.stories";

describe("AlphaColorPicker", () => {
  const configPath = path.resolve(__dirname, "../docs");
  const config = loadConfigurations(configPath);

  test("Hex", () => {
    const example = Hex;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Rgba", () => {
    const example = Rgba;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });
});
