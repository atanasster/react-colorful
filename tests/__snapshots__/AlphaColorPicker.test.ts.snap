// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlphaColorPicker Hex 1`] = `
<DocumentFragment>
  <div
    class="react-colorful"
  >
    <div
      class="react-colorful__saturation"
      style="background-color: rgb(0, 0, 0);"
    >
      <div
        aria-label="Color"
        aria-valuetext="Saturation 69%, Brightness 87%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__saturation-pointer"
          style="top: 13%; left: 69%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(69, 69, 69);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__hue"
    >
      <div
        aria-label="Hue"
        aria-valuetext="353"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__hue-pointer"
          style="top: 50%; left: 98.05555555555556%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(0, 0, 0);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__alpha react-colorful__last-control"
    >
      <div
        class="react-colorful__alpha-gradient"
      />
      <div
        aria-label="Alpha"
        aria-valuetext="100%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__alpha-pointer"
          style="top: 50%; left: 100%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(69, 69, 69);"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style="padding-top: 10px; font-weight: bold;"
  >
    #dd4455
  </div>
</DocumentFragment>
`;

exports[`AlphaColorPicker Rgba 1`] = `
<DocumentFragment>
  <div
    class="react-colorful"
  >
    <div
      class="react-colorful__saturation"
      style="background-color: rgb(0, 0, 0);"
    >
      <div
        aria-label="Color"
        aria-valuetext="Saturation 89%, Brightness 75%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__saturation-pointer"
          style="top: 25%; left: 89%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(21, 21, 21);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__hue"
    >
      <div
        aria-label="Hue"
        aria-valuetext="287"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__hue-pointer"
          style="top: 50%; left: 79.72222222222223%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(0, 0, 0);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__alpha react-colorful__last-control"
    >
      <div
        class="react-colorful__alpha-gradient"
      />
      <div
        aria-label="Alpha"
        aria-valuetext="50%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__alpha-pointer"
          style="top: 50%; left: 50%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgba(21, 21, 21, 0.5);"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style="padding-top: 10px; font-weight: bold;"
  >
    {
  "r": 154,
  "g": 22,
  "b": 192,
  "a": 0.5
}
  </div>
</DocumentFragment>
`;
